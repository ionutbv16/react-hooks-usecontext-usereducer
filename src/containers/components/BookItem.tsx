import React from 'react';
import styled from 'styled-components';
import Card from "@material-ui/core/Card";
import { makeStyles } from '@material-ui/core/styles';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { IBooks } from '../../types';

const useStyles = makeStyles({
  root: {
    maxWidth: 345,
    cursor: "default",
  },
  media: {
    height: 140,
    width: 345,
    cursor: "default",
  },
  content: {
    cursor: "default",
  }
});

const Wrapper = styled.div`
  font-size: 14px;
  flex:2;
  padding: 14px;
  display: flex;
  align-items: center;
`;

const BookItem: React.FC<IBooks> = ({ id, title, author, cover, history }) => {
  
  const classes = useStyles();

  return (
  <Wrapper>
    <Card className={classes.root}>
      <CardActionArea>
        <CardMedia
          className={classes.media}
          image={cover}
          title={title}
        />
        <CardContent className={classes.content}>
          <Typography gutterBottom variant="h5" component="h2">
          {title}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
          {author}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Button size="small" color="default" onClick={() => history.push(`/book/${id}`)}>
          More information
        </Button>
      </CardActions>
    </Card> 
  </Wrapper>
  )
  };

export default BookItem;
