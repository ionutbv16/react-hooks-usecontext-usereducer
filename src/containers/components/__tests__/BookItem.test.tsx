import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import BookItem from '../BookItem';

describe('Test BookItem Component', () => {
    it('should render without error', () => {
    const props = {
        title: 'title',
        author: 'Author',
        cover: 'https://lorempixel.com/640/480/?9d8b21dd8f74',
        id: "89cae71c-fbe5-445c-8299-6de7a88ea5ab",
    }
    const component = renderer.create(<MemoryRouter><BookItem {...props} /></MemoryRouter>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    });
});
