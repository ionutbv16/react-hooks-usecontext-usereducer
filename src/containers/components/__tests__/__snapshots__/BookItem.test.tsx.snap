// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BookItem Component should render without error 1`] = `
<div
  className="sc-bdVaJa IMKmx"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiCardMedia-root makeStyles-media-2"
        style={
          Object {
            "backgroundImage": "url(\\"https://lorempixel.com/640/480/?9d8b21dd8f74\\")",
          }
        }
        title="title"
      />
      <div
        className="MuiCardContent-root makeStyles-content-3"
      >
        <h2
          className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          title
        </h2>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          Author
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          More information
        </span>
      </button>
    </div>
  </div>
</div>
`;
